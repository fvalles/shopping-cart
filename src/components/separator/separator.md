Basic separator component with default height={1} and color="separator":

```jsx
<Separator />
```

Separator component with custom properties set:

```jsx
<Separator height={4} color="primary" />
```
