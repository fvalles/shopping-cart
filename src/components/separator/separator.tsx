import styled from 'styled-components';
import { TypographyProps } from '../typography/common';

/**
 * Types
 */

interface SeparatorProps extends Pick<TypographyProps, 'color'> {
  height?: number;
}

/**
 * Separator
 */

export const Separator = styled.div<SeparatorProps>`
  height: ${({ height = 1 }) => `${height}px`};
  border-bottom: ${({ color = 'separator', height = 1, theme }) =>
    `${height}px solid ${theme.Colors[color]}`};
`;
