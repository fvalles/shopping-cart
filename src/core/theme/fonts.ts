export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
