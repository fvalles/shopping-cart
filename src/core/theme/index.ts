export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
