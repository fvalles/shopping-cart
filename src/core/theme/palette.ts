export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
