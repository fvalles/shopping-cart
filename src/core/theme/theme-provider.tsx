import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
