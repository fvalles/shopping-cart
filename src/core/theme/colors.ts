import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
