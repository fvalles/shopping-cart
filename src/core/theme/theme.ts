import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
