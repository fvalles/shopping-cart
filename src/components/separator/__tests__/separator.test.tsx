import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Separator } from '../separator';

describe('separator component', () => {
  it('should render with default height and border-bottom', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Separator />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('height', '1px');
    expect(result).toHaveStyleRule('border-bottom', '1px solid #21224029');
    expect(result).toMatchSnapshot();
  });

  it('should render with 2px height and default border-bottom', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Separator height={2} />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('height', '2px');
    expect(result).toHaveStyleRule('border-bottom', '2px solid #21224029');
    expect(result).toMatchSnapshot();
  });
});
