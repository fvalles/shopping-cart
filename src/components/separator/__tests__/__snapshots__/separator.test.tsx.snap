// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`separator component should render with 2px height and default border-bottom 1`] = `
.c0 {
  height: 2px;
  border-bottom: 2px solid #21224029;
}

<div
  className="c0"
  height={2}
/>
`;

exports[`separator component should render with default height and border-bottom 1`] = `
.c0 {
  height: 1px;
  border-bottom: 1px solid #21224029;
}

<div
  className="c0"
/>
`;
